docs/guides/features/command.md

```diff
@@ -164,28 +164,61 @@ It also gives you full control over the way your error type gets serialized. In
 
 ## Async Commands
 
+Asynchronous functions are benefical in Tauri to perform heavy work in a manner that doesn't result in UI freezes or slowdowns.
+
 :::note
 
 Async commands are executed on a separate thread using [`async_runtime::spawn`].
 Commands without the _async_ keyword are executed on the main thread unless defined with _#[tauri::command(async)]_.
 
 :::
 
-If your command needs to run asynchronously, simply declare it as `async`:
+
+:::note
+
+You need to be careful when creating asynchronous functions using Tauri. Currently, you cannot include types that are borrowed, such as those with the & symbol, in the constructor of an asynchronous function. Some common examples of types like this are `&str` and `State<'_, Data>`.
```

Comment:
```suggestion
You need to be careful when creating asynchronous functions using Tauri. Currently, you cannot include borrowed arguments in the signature of an asynchronous function. Some common examples of types like this are `&str` and `State<'_, Data>`.
```

My main issue with this one is `such as those with the & symbol` and `constructor`. "constructor" is just wrong in this context afaik, and "such as those with the & symbol" sounds a bit too weird imo, even though i see how that would help beginners understand what we're talking about.

Reply:
Definitely, sounds a lot better than fluid. I think `constructor` is the name for function signatures in Kotlin, got my language lingo a bit mixed up there, thanks.

Reply:
Commited!